sphinx_needs/data.py

```diff
@@ -118,6 +118,7 @@ class NeedsInfoType(TypedDict, total=False):
     pre_template: Required[None | str]
     post_template: Required[None | str]
     content: Required[str]
+    content_offset: Required[int | None]
```

Comment:
We could do so, but to implement something for #1148 or #689 we have to render the content, and so should use the lino and content_offset as information for rendering. So I would go the other way around and would enable to have lineno in the needs.json.

Reply:
> would enable to have lineno in the needs.json

this is not within the current scope of the sphinx-needs / sphinx design; if you are pulling in data from a `needs.json`, then for most use cases this is from an external project (one that is different to the current sphinx project).

If sphinx-needs, starts using `docname` / `lineno` / `content_offset` from `needs.json`, then sphinx would try to report warning locations, as if they were part of the current project,
e.g. if the `docname` was `index` and the `lineno` is `1`, then sphinx would report a warning at:

```
/path/to/current/project/srcdir/index.rst::1 WARNING
```

hopefully you can see how this would be misleading?

Reply:
Even, if another tool wants to create a link to the need in the scm system (git), it would need to know where in the file the need is located, so it would need the line number.

Reply:
> > would enable to have lineno in the needs.json
> 
> this is not within the current scope of the sphinx-needs / sphinx design; if you are pulling in data from a `needs.json`, then for most use cases this is from an external project (one that is different to the current sphinx project).
> 
> If sphinx-needs, starts using `docname` / `lineno` / `content_offset` from `needs.json`, then sphinx would try to report warning locations, as if they were part of the current project, e.g. if the `docname` was `index` and the `lineno` is `1`, then sphinx would report a warning at:
> 
> ```
> /path/to/current/project/srcdir/index.rst::1 WARNING
> ```
> 
> hopefully you can see how this would be misleading?

The creation of correct warnings for external needs is something we have to solve. In all cases, we have to use the correct line number:
`/path/to/external/project/<docname><doctype>::<lineno> WARNING` and with `/path/to/external/project/` as config parameter with in [needs-external-needs](https://sphinx-needs.readthedocs.io/en/latest/configuration.html#needs-external-needs) it would be straightforward.

Reply:
> Even, if another tool wants to create a link to the need in the scm system (git), it would need to know where in the file the need is located, so it would need the line number.

this is outside the scope of this PR though, and needs to be considered separately

Reply:
Topic has now a Issue #1156